### PR TITLE
[JENKINS-68284] Modernise log recorder pages

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-68284](https://issues.jenkins.io/browse/JENKINS-68284). Possible alternative to https://github.com/jenkinsci/jenkins/pull/4815

<details>
<summary>Configure log recorder</summary>

image

</details>

<details>
<summary>Configure levels</summary>

image

</details>

Closes https://github.com/jenkinsci/jenkins/pull/4815

### Proposed changelog entries

* Modernise log recorder pages.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/hudson/logging/LogRecorder/configure.jelly
Comment (timja) [inline core/src/main/resources/hudson/logging/LogRecorder/configure.jelly:63]: changed default level to INFO so that it doesn't default to an empty FINE log recorder, alternative or at least complementary to https://github.com/jenkinsci/jenkins/pull/4815

cc @Dohbedoh

File: core/src/main/resources/hudson/logging/LogRecorder/configure.jelly
Comment (timja) [inline core/src/main/resources/hudson/logging/LogRecorder/configure.jelly:42]: downgrade to info as it looks quite poor to have a page giving warnings on load, best to avoid the default being wrong imo (fine level with empty recorder).

File: core/src/main/resources/hudson/logging/LogRecorderManager/new.jelly
Comment (timja) [inline core/src/main/resources/hudson/logging/LogRecorderManager/new.jelly:37]: added validation to match what's done on submission, error message is not very nice otherwise.

still not great that you can submit known bad input, but that's a very separate piece of work

File: core/src/main/java/hudson/logging/LogRecorderManager.java
Comment (daniel-beck) [inline core/src/main/java/hudson/logging/LogRecorderManager.java:166]:
```suggestion
    @Restricted(NoExternalUse.class)
    public FormValidation doCheckNewName(@QueryParameter String name) {
```

(or even `DoNotUse`, but that's not imported yet).

**Review by NotMyFault — APPROVED** [review]

**Review by alecharp — APPROVED** [review]
Tested with core-pr-tester and it look great.

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by basil — COMMENTED** [review]
Wow, what a huge improvement! I really admire your ability to clean up these crufty old screens. This looks great, and it feels much smoother as well. The only feedback I have is regarding the change to the default level. This does not make sense to me for the reasons explained below, and if this is undone I am happy to approve this PR.

Defaulting to `INFO` level does not make sense to me, as that is the level the JVM defaults to in [`logging.properties`](https://github.com/openjdk/jdk11u-dev/blob/e6ec27e5e1631d24070c3145126237e594fa1b5e/src/java.logging/share/conf/logging.properties#L29=) and what we would be logging to the console (or `--logfile`, if specified) in the default configuration. I think the primary use case for the **System Log** feature is to selectively enable secondary logging for specific subsystems while debugging a problem (i.e., shining a flashlight on a target), not as a replacement for the primary logging to the console (which is like a dim overhead ceiling light). That is, the way I have used this feature (and the way I imagine it is intended to be used in the primary use case) is:

1. Encounter a problem, e.g. in the Mailer plugin.
2. Check the console logs (`INFO` level) and find that they have nothing of relevance.
3. Configure a log recorder for e.g. `hudson.plugins.emailext` at a level above `INFO` to get more detail, gaining the benefit of increased visibility in a targeted area at the cost of a temporary decrease in performance and/or security in that same area.
4. Debug the problem.
5. Once the cause of the problem has been identified and the benefit of increased visibility is no longer required, disable the logger to remove the cost of performance and/or security.

With this use case, defaulting to the `INFO` level does not make sense to me, because it would always be necessary to change it in step 3 to provide any information that was not already available in the console logs. A default that has to be changed every time in the primary use case does not seem like a good default to me.

I think the original default level made sense in the context of the use case of shining a flashlight on a targeted area. the only problem was that it was all too easy to forget to fill in a specific package name or class name, thus lighting up the entire system, at a severe cost to performance and security. So that is really what we want to avoid, not the use of `FINE` level logs for a short amount of time in general – which is indeed the entire point of this feature in my opinion.

**Comment by timja** [comment]
I've got a confirmation dialog approach mostly working,

Would this be a better alternative?

image

File: core/src/main/resources/hudson/logging/LogRecorderManager/levels.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/logging/LogRecorderManager/levels.jelly:79]:
```suggestion
          <f:bottomButtonBar>
            <f:submit value="${%Save}" />
          </f:bottomButtonBar>
```

Would be nice to have the submit button stuck to the bottom for smaller screens

File: core/src/main/resources/hudson/logging/LogRecorder/configure.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/logging/LogRecorder/configure.jelly:40]: image

I don't think this 'Enter the name of the logger you want to enable' line is necessary, looks a bit odd with the notice in between it and the input.

File: core/src/main/resources/hudson/logging/LogRecorderManager/levels.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/logging/LogRecorderManager/levels.jelly:60]: Would be nice to have each input + label wrapped in its own `jenkins-form-item` or `<f:entry/>` as this would fix the spacing between the components.

**Comment by janfaracik** [comment]
Only just got the chance to properly look at this - looks great :) A few comments from me but nothing important 👍

**Review by janfaracik — APPROVED** [review]

**Comment by basil** [comment]
I would be satisfied if the default log level was left at its current setting, but I certainly would not object to a more sophisticated approach of showing a confirmation dialog. That seems ideal, though a bit more work.

**Comment by timja** [comment]
> I would be satisfied if the default log level was left at its current setting, but I certainly would not object to a more sophisticated approach of showing a confirmation dialog. That seems ideal, though a bit more work.

Updated, will possibly look at adding the confirm part in a follow-up.

**Review by basil — APPROVED** [review]

**Pull request merged**
